src/util/earthEngine.js

```diff
@@ -65,20 +64,21 @@ export const getPeriodFromFilter = (filter, datasetId) => {
         year,
     }
 }
-
 // Returns auth token for EE API as a promise
 /* eslint-disable no-async-promise-executor */
-export const getAuthToken = () =>
+export const getAuthTokenFn = (engine) => () =>
     new Promise(async (resolve, reject) => {
```

Comment:
Destructuring fails if the request fails.

```suggestion
        new Promise(async (resolve, reject) => {
    let token

    try {
        const result = await engine.query({ token: { resource: 'tokens/google' } })
        token = result.token
    } catch (error) {
        return reject(
            new Error(
                i18n.t('Cannot get authorization token for Google Earth Engine.')
            )
        )
    }

    if (token?.status === 'ERROR') {
        return reject(
            new Error(
                i18n.t(
                    'This layer requires a Google Earth Engine account. Check the DHIS2 documentation for more information.'
                )
            )
        )
    }

    resolve({
        token_type: 'Bearer',
        ...token,
    })
})
```